proposals/0033-data-collections.md

```diff
@@ -0,0 +1,244 @@
+
+# Detailed Design
+
+Data collections are created with the `defineCollection()` utility, and **must** include `type: 'data'` to store JSON files. This means "mixed" collections containing both content and data entries are not supported, and should raise a helpful error to correct.
+
+```ts
+// src/content/config.ts
+import { defineCollection, z } from 'astro:content';
+
+const authors = defineCollection({
+  type: 'data',
+  schema: z.object({
+    name: z.string(),
+    twitter: z.string().url(),
+  })
+});
+
+const blog = defineCollection({
+  // `type` can be omitted for content collections
+  // to make this change non-breaking.
+  // `type: 'content'` can also be used for completeness.
+  schema: z.object({...}),
+})
+
+export const collections = { authors };
+```
+
+## Return type
+
+Data collection entries include the same `id`, `data`, and `collection` properties as content collections:
+
+- `id (string)` - The entry file name with the extension omitted. Spaces and capitalization are preserved.
+- `collection (string)` - The collection name
+- `data (object)` - The entry data as a JS object, parsed by the configured collection schema (if any).
+
+This also means content-specific fields like `slug`, `body`, and `render()` properties are **not** included. This is due to the following:
+
+- `render()`: this function is used by content collections to parse the post body into a usable Content component. Data collections do not have HTML to render, so the function is removed.
+- `slug`: This is provided by content collections as a URL-friendly version of the file `id` for use as pages. Since data collections are not meant to be used as pages, this is omitted.
+- `body`: Unlike content collections, which feature a post body separate from frontmatter, data collections are just... data. This field could be returned as the raw JSON body, though this would give `body` a double meaning depending on the context: non-data information for content collections, and the "raw" data itself for data collections. We avoid returning the body to avoid this confusion.
+
+## Querying
+
+Today's `getCollection()` utility can be used to fetch both content or data collections:
+
+```astro
+---
+import { getCollection } from 'astro:content';
+
+const authors = await getCollection('authors');
+---
+
+<h1>Our Authors</h1>
+<ul>
+  {authors.map(author => (
+    <li><a href={author.data.twitter}>{author.data.name}</a></li>
+  ))}
+</ul>
+```
+
+To retrieve individual entries, `getEntry()` can be used. This receives both the collection name and the entry `id` as described in the [Return type](#return-type). These can be passed as separate arguments or as object keys.
```

Comment:
Huge fan of this new `getEntry` API! It's a clever solution to the ID vs. slug naming complexity you outline below.